packages/server/src/traceparent.ts

```diff
@@ -0,0 +1,30 @@
+// https://www.w3.org/TR/trace-context/#traceparent-header
+
+type hex = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | 'a' | 'b' | 'c' | 'd' | 'e' | 'f';
+type twohex = `${hex}${hex}`;
+
+export type Traceparent =
+  | { valid: false }
+  | {
+      valid: true;
+      version: twohex | null;
+      traceId: string;
+      parentId: string;
+      flags: twohex | null;
```

Comment:
`twohex` is an invalid typing for `flags` as the regular expression allows 1 character in it's current form